etna/models/prophet.py

```diff
@@ -152,6 +154,76 @@ def predict(self, df: pd.DataFrame, prediction_interval: bool, quantiles: Sequen
         y_pred = y_pred.rename(rename_dict, axis=1)
         return y_pred
 
+    def _check_mul_components(self, ignore_components: Set[str]):
+        """Raise error if model contains multiplicative components."""
+        components_modes = self.model.component_modes
+        if components_modes is None:
+            raise ValueError("This model is not fitted!")
+
+        mul_components = set(self.model.component_modes["multiplicative"])
+        if len(mul_components - ignore_components) > 0:
+            raise ValueError("Forecast decomposition is only supported for additive components!")
+
+    def _predict_seasonal_components(self, df: pd.DataFrame, ignore_components: Set[str]) -> pd.DataFrame:
+        """Estimate seasonal, holidays and exogenous components."""
+        model = self.model
+
+        seasonal_features, _, component_cols, _ = model.make_all_seasonality_features(df)
+
+        holiday_names = set(model.train_holiday_names) if model.train_holiday_names is not None else set()
+
+        components_data = {}
+        for component_name in component_cols.columns:
+            if component_name in ignore_components or component_name in holiday_names:
+                continue
+
+            beta_c = model.params["beta"] * component_cols[component_name].values
+            comp = seasonal_features.values @ beta_c.T
+
+            # apply rescaling for additive components
+            comp *= model.y_scale
+
+            components_data[component_name] = bn.nanmean(comp, axis=1)
+
+        return pd.DataFrame(data=components_data)
+
+    def predict_components(self, df: pd.DataFrame) -> pd.DataFrame:
+        """Estimate prediction components.
+
+        Parameters
+        ----------
+        df:
+            features dataframe
+
+        Returns
+        -------
+        :
+            dataframe with prediction components
+        """
+        # aggregation of corresponding model terms, e.g. sum
+        aggregated_components = {
+            "additive_terms",
+            "multiplicative_terms",
+            "extra_regressors_additive",
+            "extra_regressors_multiplicative",
+        }
+
+        self._check_mul_components(aggregated_components)
+
+        df = df.reset_index()
```

Comment:
May be we can wrap this block into the separate method `_prepare_prophet_df` as it is also used in `predict`